### PR TITLE
Binary publish

### PR DESCRIPTION
**Pull request by HamidShojanazeri**
This PR adds scripts for building binaries and publishing to pypi.

### CODE REVIEW

File: binaries/build.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    Function to build the wheel files for torchserve, model-archiver and workflow-archiver
+    """
+    if args.nightly:
+        print(
+            "## Started pippy build"
+        )
+        create_wheel_cmd = "python setup.py "
+    else:
+        print("## Started pippy build")
+        create_wheel_cmd = "python setup.py bdist_wheel "
```
Comment (lxning) [inline binaries/build.py:22]: is "--release" needed at here?
create_wheel_cmd = "python setup.py bdist_wheel " =>
create_wheel_cmd = "python setup.py bdist_wheel --release"

File: binaries/build.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+def build(args):
+
+    # Build dist wheel files
+    build_dist_whl(args)
+
+    os.chdir(REPO_ROOT)
+
+    if not args.dry_run:
+        ts_wheel_path = glob.glob(os.path.join(REPO_ROOT, "dist", "*.whl"))[0]
```
Comment (lxning) [inline binaries/build.py:52]: change "ts_wheel_path" to "pippy_wheel_path"?

File: binaries/build.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    else:
+        ts_wheel_path = os.path.join(REPO_ROOT, "dist", "*.whl")
```
Comment (lxning) [inline binaries/build.py:55]: Ditto

File: binaries/upload.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Upload anaconda and pypi packages for torchserve and torch-model-archiver"
+    )
+    parser.add_argument(
+        "--upload-conda-packages",
+        action="store_true",
+        required=False,
+        help="Specify whether to upload conda packages",
+    )
```
Comment (lxning) [inline binaries/upload.py:46]: no conda

File: setup.py [hunk 5]
```diff
@@ -67,6 +72,14 @@ def run(self):
...
 if __name__ == "__main__":
     write_version_file()
+    NAME_ARG = "--override-name"
+    if NAME_ARG in sys.argv:
+        idx = sys.argv.index(NAME_ARG)
+        name = sys.argv.pop(idx + 1)
+        sys.argv.pop(idx)
+        is_nightly = "nightly" in name
+        if is_nightly:
+            version = get_nightly_version()
```
Comment (kwen2501) [inline setup.py:82]: I wonder if we'd need a nightly package yet?

Per the command in `build.py`, this code section would introduce another package named `PiPPy-nightly` (case-sensitive, defined in `build.py`), as compared to `pippy` (defined in this file).

File: binaries/upload.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        "--upload-pypi-packages",
+        action="store_true",
+        required=False,
+        help="Specify whether to upload pypi packages",
+    )
+    parser.add_argument(
+        "--test-pypi",
+        action="store_true",
+        required=False,
+        help="Specify whether to upload to test PyPI",
```
Comment (kwen2501) [inline binaries/upload.py:52]: Duplicated help description?

File: binaries/upload.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+                f"twine upload {dist_path}/* --username __token__ --repository-url https://test.pypi.org/legacy/",
+                dry_run,
+            )
+        else:
+            try_and_handle(f"twine upload --username __token__ {dist_path}/*", dry_run)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Upload anaconda and pypi packages for torchserve and torch-model-archiver"
```
Comment (kwen2501) [inline binaries/upload.py:39]: `torchserve and torch-model-archiver` --> `pippy`

**Review by kwen2501 — COMMENTED** [review]
Thanks MUCH for writing the build scripts!

I wonder if you could write down some example commands in the PR description so that we can test it? Thanks!

**Review by seemethere — APPROVED** [review]

File: binaries/upload.py [hunk 8]
```diff
@@ -45,12 +45,6 @@ def upload_pypi_packages(args, WHL_PATHS):
         required=False,
         help="Specify whether to upload pypi packages",
     )
-    parser.add_argument(
-        "--test-pypi",
-        action="store_true",
-        required=False,
-        help="Specify whether to upload to test PyPI",
-    )
```
Comment (kwen2501) [inline binaries/upload.py]: It seems the wrong section is deleted.
Maybe we want to delete `--upload-pypi-packages` (not used) and keep `--test-pypi` (used)?

**Pull request merged**